web/src/message_list_view.ts

```diff
@@ -159,6 +161,16 @@ function same_recipient(a: MessageContainer | undefined, b: MessageContainer | u
     return util.same_recipient(a.msg, b.msg);
 }
 
+function is_search_view(): boolean {
+    const path = window.location.hash.split("/");
+    // The following check ensures the user is actually searching for something,
+    // and not to confuse it with the case in which the channel or topic name is "search".
+    if (path.length >= 2 && path.at(-2) === "search") {
+        return true;
+    }
```

Comment:
### This matches the following cases : 

- `/#narrow/search/foo`
- `/#narrow/stream/stream_name/search/foo`
- `/#narrow/stream/search/topic/search/search/foo` 
- `/#narrow/stream/search/topic/search/search/search` both channel and topic are named "search", and the user is searching the string "search"

### But doesn't match the following : 

- `/#narrow/stream/stream_name/topic/search`
- `/#narrow/stream/search/topic/search`

Reply:
Use a function in `filter.ts` to check this, rather than ad-hoc parsing the URL in another place.

Reply:
I see `filter.ts` responsible for the searching functionality but nothing there checks if current view (e.g. current URL) is a search view

Reply:
Instead I will move my function to `hash_util.ts`